src/cljs/main/broadfcui/common/table/utils.cljs

```diff
@@ -43,7 +43,8 @@
     data))
 
 (defn- matches-filter-text [filter-tokens source]
-  (let [lc-source (string/lower-case source)]
+  (assert (string? source) (str "Attempting to filter table row text that is not a string: " source)) ; GAWB-2872
```

Comment:
Are you throwing an exception here to help find the problem? I'm curious what `source` is that isn't a string and I think it's worth finding out. However, if CLJS exceptions are like in other languages, this will prevent `string/lower-case` from being called on anything other than a string so the `(str ...)` change below won't change anything.

Reply:
> this will prevent string/lower-case from being called on anything other than a string
Correct, this is what we want - the tests found a bug, and knowing what the bad data is will help us fix it.

Asserts are compiled out in production, so the `str` protects users from (console) JS errors.

Reply:
Ahh, good point. I knew they were suppressed somehow but didn't know they were compiled out.

The other option would be to not match if `source` is not a string. Since we don't know what it is when it's not a string, we don't know if it's appropriate to match on its string representation.